src/dataframe/dataframe.jl

```diff
@@ -466,38 +466,65 @@ function Base.setindex!(df::DataFrame, v::Any, row_ind::Real, col_ind::ColumnInd
     insert_single_entry!(df, v, row_ind, col_ind)
 end
 
-# df[SingleRowIndex, MultiColumnIndex] = Single Item
+# df[SingleRowIndex, :] = Union{AbstractDict, NamedTuple}
 function Base.setindex!(df::DataFrame,
-                        v::Any,
-                        row_ind::Real,
-                        col_inds::AbstractVector{Bool})
-    setindex!(df, v, row_ind, findall(col_inds))
-end
-function Base.setindex!(df::DataFrame,
-                        v::Any,
+                        row::Union{AbstractDict, NamedTuple},
                         row_ind::Real,
-                        col_inds::AbstractVector{<:ColumnIndex})
-    for col_ind in col_inds
-        insert_single_entry!(df, v, row_ind, col_ind)
+                        ::Colon)
+    old_row = df[row_ind, :]
```

Comment:
This will create `DataFrameRow` so we cannot use it to handle cleanup.

Reply:
Indeed. So I guess the best solution is to call `copy`?

Reply:
Right point.